modules/dublincore/src/main/java/org/opencastproject/metadata/dublincore/MetadataField.java

```diff
@@ -1041,60 +1038,84 @@ public static JObject mapToJSON(Map<String, String> map) {
+            ? dublinCoreProperties.get(CONFIG_LABEL_KEY) : null;
+    Opt<String> listprovider = dublinCoreProperties.containsKey(CONFIG_LIST_PROVIDER_KEY)
+            ? Opt.some(dublinCoreProperties.get(CONFIG_LIST_PROVIDER_KEY)) : Opt.none();
+    Opt<String> namespace = dublinCoreProperties.containsKey(CONFIG_NAMESPACE_KEY)
+            ? Opt.some(dublinCoreProperties.get(CONFIG_NAMESPACE_KEY)) : Opt.none();
+    Type type = dublinCoreProperties.containsKey(CONFIG_TYPE_KEY)
+            ? Type.valueOf(dublinCoreProperties.get(CONFIG_TYPE_KEY).toUpperCase()) : null;
+    Boolean required = dublinCoreProperties.containsKey(CONFIG_REQUIRED_KEY)
+            ? Boolean.valueOf(dublinCoreProperties.get(CONFIG_REQUIRED_KEY).toUpperCase()) : null;
+    Boolean readOnly = dublinCoreProperties.containsKey(CONFIG_READ_ONLY_KEY)
+            ? Boolean.valueOf(dublinCoreProperties.get(CONFIG_READ_ONLY_KEY).toUpperCase()) : null;
+    Opt<String> outputID = dublinCoreProperties.containsKey(CONFIG_OUTPUT_ID_KEY)
+            ? Opt.some(dublinCoreProperties.get(CONFIG_OUTPUT_ID_KEY)) : Opt.none();
+    Opt<Integer> order = Opt.none();
+    if (dublinCoreProperties.containsKey(CONFIG_ORDER_KEY)) {
+      try {
+        order = Opt.some(Integer.parseInt(dublinCoreProperties.get(CONFIG_ORDER_KEY)));
+      } catch (NumberFormatException e) {
+        logger.warn("Unable to parse order value {} of metadata field {}", dublinCoreProperties.get(CONFIG_ORDER_KEY),
+                inputID, e);
+      }
+    }
+
+    MetadataField metadataField = createMetadataField(inputID, outputID, label, readOnly, required, Opt.none(), type,
+            Opt.none(), collectionID, order, namespace, delimiter, pattern);
+    metadataField.setListprovider(listprovider);
+    return metadataField;
+  }
+
+  public static MetadataField createMetadataField(String inputID, Opt<String> outputID, String label, boolean readOnly,
+          boolean required, Opt<Boolean> translatable, Type type, Opt<Map<String, String>> collection,
+          Opt<String> collectionID, Opt<Integer> order, Opt<String> namespace, Opt<String> delimiter, String pattern) {
+
+    switch (type) {
+      case BOOLEAN:
+        return createBooleanMetadata(inputID, outputID, label, readOnly, required, order, namespace);
+      case DATE:
+        return createDateMetadata(inputID, outputID, label, readOnly, required, pattern, order,
+                namespace);
+      case DURATION:
+        return createDurationMetadataField(inputID, outputID, label, readOnly, required, translatable,
+                collection, collectionID, order, namespace);
+      case ITERABLE_TEXT:
+        // Add an iterable text style field
+        return createIterableStringMetadataField(inputID, outputID, label, readOnly, required, translatable,
+                        collection, collectionID, delimiter, order, namespace);
+      case MIXED_TEXT:
+        // Add an iterable text style field
```

Comment:
Two times "iterable text style field"? I suggest to just remove those two comments as the method names already tell you which type of metadata field you add.

Reply:
Fixed.